src/App.tsx

```diff
@@ -1,62 +1,117 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
+import cn from 'classnames';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+enum SortType {
+  Dumplings = 'Dumplings',
+  Carrot = 'Carrot',
+  Eggs = 'Eggs',
+  IceCream = 'Ice cream',
+  Apple = 'Apple',
+  Bread = 'Bread',
+  Fish = 'Fish',
+  Honey = 'Honey',
+  Jam = 'Jam',
+  Garlic = 'Garlic',
+}
+
+function getPreparedGoods(
+  goods: SortType[],
+  sortField: string,
+  isReversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
 export const App: React.FC = () => {
+  const valuesSortType = Object.values(SortType);
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(valuesSortType, sortField, isReversed);
+  const isResetVisible = sortField || isReversed;
+  const handleReset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+          className={cn(
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET },
```

Comment:
```suggestion
          onClick={() => setSortField(enumName.keyName)}
          className={cn(
            { 'is-light': sortField !== enumName.keyName },
```
do it with second button too